### PR TITLE
Add ADS use case in documentation

### PR DESCRIPTION
**Pull request by mariehbourget**
#### GitHub

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [ ] I've linked relevant issues in the PR body
- [x] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [x] I've assigned a reviewer

#### PR contents

- [x] I've consulted [ivadomed's internal developer documentation](https://github.com/ivadomed/ivadomed/wiki) to ensure my contribution is in line with any relevant design decisions
- [ ] I've added [relevant tests](https://github.com/ivadomed/ivadomed/wiki/tests) for my contribution
- [x] I've updated the [relevant documentation](https://github.com/ivadomed/ivadomed/wiki/documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description
This PR adds a use-case for AxonDeepSeg in documentation.

TODO before merge: Update the figure link to ivadomed/doc-figures/main/use_cases/ADS.png

## Linked issues
No related issue

### CODE REVIEW

File: docs/source/use_cases.rst [hunk 1]
```diff
@@ -6,13 +6,13 @@ Use case #1 - Spinal Cord Toolbox:
 
 `Spinal cord toolbox <http://spinalcordtoolbox.com/>`__ (SCT) is an open-source analysis software package for processing MRI data of the spinal cord `[De Leener et al. 2017] <https://doi.org/10.1016/j.neuroimage.2016.10.009>`__. `ivadomed` is SCT's backbone for the automated segmentation of the spinal cord, gray matter, tumors, and multiple sclerosis lesions, as well as for the labeling of intervertebral discs.
 
-Use case 2 - Creation of anatomical template:
----------------------------------------------
+Use case #2 - Creation of anatomical template:
```
Comment (jcohenadad) [inline docs/source/use_cases.rst:9]: no need for "use case 1", etc.
also please remove the ":"

Comment (mariehbourget) [reply docs/source/use_cases.rst]: Fixed in [99aa381](https://github.com/ivadomed/ivadomed/pull/1080/commits/99aa38104f7e2777315b0ea04e3cb871938838a9)

**Comment by mariehbourget** [comment]
This PR is ready for review. It adds a use case for AxonDeepSeg in ivadomed documentation with minor formatting updates to the use cases page.

Note to the reviewer: if approved, please also approve this PR https://github.com/ivadomed/doc-figures/pull/6 in the doc-figure repo so I can update the link to the figure to the permanent one before merging. Thanks!

**Review by hermancollin — APPROVED** [review]
LGTM. Also, for the doc-figures repo, it seems you don't need a review to merge.

**Pull request merged**